clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/DelegatingStorageEngine.java

```diff
@@ -32,6 +36,46 @@ public void setDelegate(@Nonnull StorageEngine<P> delegate) {
     this.delegate = Objects.requireNonNull(delegate);
   }
 
+  public boolean isKeyUrnCompressionEnabled(int partitionId) {
+    if (keyDictCompressionFunction == null) {
+      return false;
+    }
+    return keyDictCompressionFunction.apply(partitionId) != null;
+  }
+
+  public void setKeyDictCompressionFunction(Function<Integer, KeyUrnCompressor> keyDictCompressionFunction) {
+    this.keyDictCompressionFunction = Objects.requireNonNull(keyDictCompressionFunction);
+    if (delegate instanceof DelegatingStorageEngine) {
```

Comment:
I am a bit confused here - is it even possible to have this recursive delegate SE here?

Reply:
yes in unit test.
`DeepCopyStorageEngine` inherits `DelegatingStorageEngine`.
Maybe I should add some javadoc here to avoid the confusion.